python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,483 @@
+                                 "x": pa.int64(),
+                                 },
+                                pa.int64())
+    return binary_function, func_name
+
+
+@pytest.fixture(scope="session")
+def ternary_func_fixture():
+    def ternary_function(ctx, m, x, c):
+        mx = pc.call_function("multiply", [m, x])
+        return pc.call_function("add", [mx, c])
+    ternary_doc = {"summary": "y=mx+c",
+                   "description": "find y from y = mx + c"}
+    func_name = "y=mx+c"
+    pc.register_scalar_function(ternary_function,
+                                func_name,
+                                ternary_doc,
+                                {
+                                    "array1": pa.int64(),
+                                    "array2": pa.int64(),
+                                    "array3": pa.int64(),
+                                },
+                                pa.int64())
+    return ternary_function, func_name
+
+
+@pytest.fixture(scope="session")
+def varargs_func_fixture():
+    def varargs_function(ctx, *values):
+        base_val = values[:2]
+        res = pc.call_function("add", base_val)
+        for other_val in values[2:]:
+            res = pc.call_function("add", [res, other_val])
+        return res
+    func_name = "z=ax+by+c"
+    varargs_doc = {"summary": "z=ax+by+c",
+                   "description": "find z from z = ax + by + c"
+                   }
+    pc.register_scalar_function(varargs_function,
+                                func_name,
+                                varargs_doc,
+                                {
+                                    "array1": pa.int64(),
+                                    "array2": pa.int64(),
+                                    "array3": pa.int64(),
+                                    "array4": pa.int64(),
+                                    "array5": pa.int64(),
+                                },
+                                pa.int64())
+    return varargs_function, func_name
+
+
+@pytest.fixture(scope="session")
+def random_with_udf_ctx_func_fixture():
+    def random_with_udf_ctx(context, one, two):
+        proxy_pool = pa.proxy_memory_pool(context.memory_pool)
+        ans = pc.add(one, two, memory_pool=proxy_pool)
+        res = pa.array([ans.as_py()], memory_pool=proxy_pool)
+        return res
+    in_types = {"one": pa.int64(),
+                "two": pa.int64(),
+                }
+    func_doc = {
+        "summary": "test udf context",
+        "description": "udf context test"
+    }
+    func_name = "test_udf_context"
+    pc.register_scalar_function(random_with_udf_ctx,
+                                func_name, func_doc,
+                                in_types,
+                                pa.int64())
+    return random_with_udf_ctx, func_name
+
+
+def const_return(ctx, scalar):
+    return 42
+
+
+@pytest.fixture(scope="session")
+def output_check_func_fixture():
```

Comment:
Can you add a comment explaining what the point of this example is?

Reply:
Added a description.